In [30]:
import pandas as pd
path = "C:\\Users\\foucardguimax\\Desktop\\PortFolio\\dataset.xlsx"
df = pd.read_excel(path)
df.head()

Patient ID  Patient age quantile SARS-Cov-2 exam result  \
0  44477f75e8169d2                    13               negative   
1  126e9dd13932f68                    17               negative   
2  a46b4402a0e5696                     8               negative   
3  f7d619a94f97c45                     5               negative   
4  d9e41465789c2b5                    15               negative   

   Patient addmited to regular ward (1=yes, 0=no)  \
0                                               0   
1                                               0   
2                                               0   
3                                               0   
4                                               0   

   Patient addmited to semi-intensive unit (1=yes, 0=no)  \
0                                                  0       
1                                                  0       
2                                                  0       
3                                                  0       
4                                                  0       

   Patient addmited to intensive care unit (1=yes, 0=no)  Hematocrit  \
0                                                  0             NaN   
1                                                  0        0.236515   
2                                                  0             NaN   
3                                                  0             NaN   
4                                                  0             NaN   

   Hemoglobin  Platelets  Mean platelet volume   ...  \
0         NaN        NaN                    NaN  ...   
1    -0.02234  -0.517413               0.010677  ...   
2         NaN        NaN                    NaN  ...   
3         NaN        NaN                    NaN  ...   
4         NaN        NaN                    NaN  ...   

   Hb saturation (arterial blood gases)  pCO2 (arterial blood gas analysis)  \
0                                   NaN                                 NaN   
1                                   NaN                                 NaN   
2                                   NaN                                 NaN   
3                                   NaN                                 NaN   
4                                   NaN                                 NaN   

   Base excess (arterial blood gas analysis)  \
0                                        NaN   
1                                        NaN   
2                                        NaN   
3                                        NaN   
4                                        NaN   

   pH (arterial blood gas analysis)  Total CO2 (arterial blood gas analysis)  \
0                               NaN                                      NaN   
1                               NaN                                      NaN   
2                               NaN                                      NaN   
3                               NaN                                      NaN   
4                               NaN                                      NaN   

   HCO3 (arterial blood gas analysis)  pO2 (arterial blood gas analysis)  \
0                                 NaN                                NaN   
1                                 NaN                                NaN   
2                                 NaN                                NaN   
3                                 NaN                                NaN   
4                                 NaN                                NaN   

   Arteiral Fio2  Phosphor  ctO2 (arterial blood gas analysis)  
0            NaN       NaN                                 NaN  
1            NaN       NaN                                 NaN  
2            NaN       NaN                                 NaN  
3            NaN       NaN                                 NaN  
4            NaN       NaN                                 NaN  

[5 rows x 111 columns]

In [31]:
print(df.shape)

(5644, 111)


Gros data set, il y a 111 colonnes (features) possiblement exploitables et 5644 lignes (samples). On va affichier ces colonnes.

In [32]:
list_col = df.columns
for feature in list_col:
    print(feature)

Patient ID
Patient age quantile
SARS-Cov-2 exam result
Patient addmited to regular ward (1=yes, 0=no)
Patient addmited to semi-intensive unit (1=yes, 0=no)
Patient addmited to intensive care unit (1=yes, 0=no)
Hematocrit
Hemoglobin
Platelets
Mean platelet volume 
Red blood Cells
Lymphocytes
Mean corpuscular hemoglobin concentration (MCHC)
Leukocytes
Basophils
Mean corpuscular hemoglobin (MCH)
Eosinophils
Mean corpuscular volume (MCV)
Monocytes
Red blood cell distribution width (RDW)
Serum Glucose
Respiratory Syncytial Virus
Influenza A
Influenza B
Parainfluenza 1
CoronavirusNL63
Rhinovirus/Enterovirus
Mycoplasma pneumoniae
Coronavirus HKU1
Parainfluenza 3
Chlamydophila pneumoniae
Adenovirus
Parainfluenza 4
Coronavirus229E
CoronavirusOC43
Inf A H1N1 2009
Bordetella pertussis
Metapneumovirus
Parainfluenza 2
Neutrophils
Urea
Proteina C reativa mg/dL
Creatinine
Potassium
Sodium
Influenza B, rapid test
Influenza A, rapid test
Alanine transaminase
Aspartate transaminase
Gamma-glutamyltransfe

In [33]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5644 entries, 0 to 5643
Columns: 111 entries, Patient ID to ctO2 (arterial blood gas analysis)
dtypes: float64(70), int64(4), object(37)
memory usage: 4.8+ MB


On voit ici que 70 des colonnes sont des numerical data, possiblement 4 sont des categorical mais arrangés de façon numérique, et 37 sont possiblement des categorical.

In [34]:
print(df.describe(include='all'))

             Patient ID  Patient age quantile SARS-Cov-2 exam result  \
count              5644           5644.000000                   5644   
unique             5644                   NaN                      2   
top     a34d12251a0e800                   NaN               negative   
freq                  1                   NaN                   5086   
mean                NaN              9.318391                    NaN   
std                 NaN              5.777903                    NaN   
min                 NaN              0.000000                    NaN   
25%                 NaN              4.000000                    NaN   
50%                 NaN              9.000000                    NaN   
75%                 NaN             14.000000                    NaN   
max                 NaN             19.000000                    NaN   

        Patient addmited to regular ward (1=yes, 0=no)  \
count                                      5644.000000   
unique             

On voit ici que tout n'est pas affiché, on va essayé de voir pour toutes les colonnes le nombre de données renseignées (ie, le nbr de missing values).

In [35]:
description = df.describe(include='all') # C'est un dataset
for i in range(description.shape[1]):
    print(str(list_col[i]) + " --> " + str(description.iloc[0,i]))

Patient ID --> 5644
Patient age quantile --> 5644.0
SARS-Cov-2 exam result --> 5644
Patient addmited to regular ward (1=yes, 0=no) --> 5644.0
Patient addmited to semi-intensive unit (1=yes, 0=no) --> 5644.0
Patient addmited to intensive care unit (1=yes, 0=no) --> 5644.0
Hematocrit --> 603.0
Hemoglobin --> 603.0
Platelets --> 602.0
Mean platelet volume  --> 599.0
Red blood Cells --> 602.0
Lymphocytes --> 602.0
Mean corpuscular hemoglobin concentration (MCHC) --> 602.0
Leukocytes --> 602.0
Basophils --> 602.0
Mean corpuscular hemoglobin (MCH) --> 602.0
Eosinophils --> 602.0
Mean corpuscular volume (MCV) --> 602.0
Monocytes --> 601.0
Red blood cell distribution width (RDW) --> 602.0
Serum Glucose --> 208.0
Respiratory Syncytial Virus --> 1354
Influenza A --> 1354
Influenza B --> 1354
Parainfluenza 1 --> 1352
CoronavirusNL63 --> 1352
Rhinovirus/Enterovirus --> 1352
Mycoplasma pneumoniae --> 0.0
Coronavirus HKU1 --> 1352
Parainfluenza 3 --> 1352
Chlamydophila pneumoniae --> 1352
Adenovirus

Visiblement il y a des colonnes sans valeurs à l'intérieur, on va les supprimer.

In [36]:
for i in range(df.shape[1]):
    if description.iloc[0,i] == 0.0:
        df = df.drop(str(list_col[i]), axis = 1)
        print(list_col[i])
print(df.shape)

Mycoplasma pneumoniae
Urine - Sugar
Partial thromboplastin time (PTT) 
Prothrombin time (PT), Activity
D-Dimer
(5644, 106)


Ok on à supprimé ces colonnes car elle n'ont aucune valeur à l'intérieur. On va maintenant regrouper les colonnes par nombre de features.

In [61]:
dic = {}
nbr_rows = description.iloc[0,:]
for i in range(nbr_rows.shape[0]):
    if nbr_rows.iloc[i] not in dic:
        dic[nbr_rows.iloc[i]] = []
    dic[nbr_rows.iloc[i]].append(nbr_rows.index[i])
dic

{5644: ['Patient ID',
  'Patient age quantile',
  'SARS-Cov-2 exam result',
  'Patient addmited to regular ward (1=yes, 0=no)',
  'Patient addmited to semi-intensive unit (1=yes, 0=no)',
  'Patient addmited to intensive care unit (1=yes, 0=no)'],
 603.0: ['Hematocrit', 'Hemoglobin'],
 602.0: ['Platelets',
  'Red blood Cells',
  'Lymphocytes',
  'Mean corpuscular hemoglobin concentration\xa0(MCHC)',
  'Leukocytes',
  'Basophils',
  'Mean corpuscular hemoglobin (MCH)',
  'Eosinophils',
  'Mean corpuscular volume (MCV)',
  'Red blood cell distribution width (RDW)'],
 599.0: ['Mean platelet volume '],
 601.0: ['Monocytes'],
 208.0: ['Serum Glucose'],
 1354: ['Respiratory Syncytial Virus', 'Influenza A', 'Influenza B'],
 1352: ['Parainfluenza 1',
  'CoronavirusNL63',
  'Rhinovirus/Enterovirus',
  'Coronavirus HKU1',
  'Parainfluenza 3',
  'Chlamydophila pneumoniae',
  'Adenovirus',
  'Parainfluenza 4',
  'Coronavirus229E',
  'CoronavirusOC43',
  'Inf A H1N1 2009',
  'Bordetella pertussis',


True


[4, 5, 6]

In [58]:
nbr_rows.index[0]

'Patient ID'